### DISCUSSION TITLE
.bin compile fails when Device_Name and Friendly_Name data is not enclosed in brackets

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#11707] [Q&A] [answered]
Used https://sapphire-landfowl-ar78ijxh.ws-us03.gitpod.io/#/workspace/tasmocompiler-new to compile a .bin file. There was problem when it came to following lines:

#define DEVICE_NAME       “First Room”       // only on development branch v 9.3.1.2 and above
#define FRIENDLY_NAME     “First Room Switch”

The compiler gave an error that it was expecting ')' before Room.

So I changed the lines to following:

#define DEVICE_NAME       “(First Room)”       // only on development branch v 9.3.1.2 and above
#define FRIENDLY_NAME     “(First Room Switch)”

and was able to successfully compile the .bin file. but the problem is that both the device name and all friendly names are enclosed in brackets. (see the attached image)

Is this a bug or format of entering the fields in user_config_override.h has changed to require brackets?

Regards,

Navi

![Screenshot (94)](https://user-images.githubusercontent.com/69470381/114532230-44393b80-9c6a-11eb-9406-a7e55f1b1409.png)

**Comment by barbudor** [comment, marked as answer]
It was building very fine without bracket yesterday before I did the PR

**Reply by NsinghP** [reply]
Here are the errors I am getting:
...
/tmp/Tasmota/tasmota/settings.ino:752:41: note: in expansion of macro 'PSTR'
   SettingsUpdateText(SET_FRIENDLYNAME4, PSTR(FRIENDLY_NAME"4"));
                                         ^
tasmota/user_config_override.h:390:36: error: expected ')' before 'Room'
 #define DEVICE_NAME       “First Room”       // only on development branch v 9.3.1.2 and above
...
...
/tmp/Tasmota/tasmota/xdrv_01_webserver.ino: In function 'void HandleOtherConfiguration()':
tasmota/user_config_override.h:391:30: error: expected ')' before 'First'
 #define FRIENDLY_NAME     “First Room Switch”

I am using following custom strings:

#define USE_UFILESYS
#define GUI_TRASH_FILE
#define MODULE            USER_MODULE
#define USER_TEMPLATE "{\"NAME\":\"Hazel\",\"GPIO\":[256,259,32,163,257,258,1,1,161,160,162,1,1,1],\"FLAG\":0,\"BASE\":18}"
#define EMULATION         EMUL_HUE
#define USE_EMULATION_HUE
#define DEVICE_NAME       “First Room”       // only on development branch v 9.3.1.2 and above
#define FRIENDLY_NAME     “First Room Switch”
#define TIME_DST_OFFSET    +330          // Offset from UTC in minutes (-780 to +780)
#define TIME_STD_OFFSET    +330      // Offset from UTC in minutes (-780 to +780)
#define APP_TIMEZONE       99    // [Timezone](-13 .. 14 = hrs UTC, 99 = use TIME_DST/TIME_STD)

It works fine if I change these two lines:
#define DEVICE_NAME       “(First Room)”       // only on development branch v 9.3.1.2 and above
#define FRIENDLY_NAME     “(First Room Switch)”

PS: I used tasmocompiler at https://moccasin-tarantula-k8fdgpkm.ws-us03.gitpod.io/#/workspace/tasmocompiler and wemos/nodemcu board.


Regards,

Navi
                              ^

**Reply by barbudor** [reply]
The problem MAY BE the comments
I'm checking that now

**Reply by barbudor** [reply]
So no problems even with comments on the define line
I can build without any error

**Reply by barbudor** [reply]
This is wrong :
```
#define USER_TEMPLATE "{"NAME":"Hazel","GPIO":[256,259,32,163,257,258,1,1,161,160,162,1,1,1],"FLAG":0,"BASE":18}"
```
You are missing a `\` for each `"` inside the string

**Reply by NsinghP** [reply]
#define USE_UFILESYS
#define GUI_TRASH_FILE
#define MODULE            USER_MODULE
#define USER_TEMPLATE "{\"NAME\":\"Hazel\",\"GPIO\":[256,259,32,163,257,258,1,1,161,160,162,1,1,1],\"FLAG\":0,\"BASE\":18}"
#define EMULATION         EMUL_HUE
#define USE_EMULATION_HUE
#define DEVICE_NAME       “First Room”  
#define FRIENDLY_NAME     “First Room Switch”
#define TIME_DST_OFFSET    +330          
#define TIME_STD_OFFSET    +330      
#define APP_TIMEZONE       99    

But get this error:

/tmp/Tasmota/tasmota/xdrv_01_webserver.ino: In function 'void HandleOtherConfiguration()':
tasmota/user_config_override.h:391:30: error: expected ')' before 'First'
 #define FRIENDLY_NAME     “First Room Switch”

**Reply by NsinghP** [reply]
\ are not missing on, they just go missing when copy and paste

![Screenshot (95)](https://user-images.githubusercontent.com/69470381/114540057-bb72cd80-9c72-11eb-89c4-39a838653db2.png)

**Reply by NsinghP** [reply]
when i remove both 

#define DEVICE_NAME “First Room”
#define FRIENDLY_NAME “First Room Switch”

it works fine

**Reply by barbudor** [reply]
May be a problem with TasmoCompiler 
@benzino77 what do you think ?

**Reply by NsinghP** [reply]
Maybe a problem on their end or it could be my browser cache, let me try with MS edge.

**Reply by sfromis** [reply]
You need to enclose the string in "real" quotation marks,
`#define DEVICE_NAME "First Room"`
Instead of
`#define DEVICE_NAME “First Room”`

**Reply by barbudor** [reply]
Good eyes @sfromis
I missed it was curved ones and not straight ones

**Reply by benzino77** [reply]
"Custom parameters" are written at the end of `user_config_override.h` file as is. I do not transform it in any way.

**Reply by benzino77** [reply]
Oh! now I see @sformis post about quotation marks... Maybe I should transform "Custom parameters" before writing them to `user_config_override.h` file. It shouldn't be hard to implement. I will put it on my "to do" list.

**Reply by NsinghP** [reply]
Hey, Big Thanks to everyone for their time. It worked, I knew quotes can be powerful motivators at times, but I didn't know the power of quotation marks!! :) I dont know how the fake quotation marks sneaked into my strings, maybe It happend when I copied the strings from sample file somewhere. 

![Screenshot (98)](https://user-images.githubusercontent.com/69470381/114549114-3d1c2880-9c7e-11eb-8c7d-ea04b3152016.png)

How come, it still shows "Device Name (First Switch)" above Device Name manual text entry area? Can this be changed to show (First Room)?

**Reply by sfromis** [reply]
The thing shown in () is normally a standard value, which you should expect to get if you erase the current content of the field.
